src/mono/mono/metadata/class-init.c

```diff
@@ -1608,6 +1608,7 @@ mono_class_create_fnptr (MonoMethodSignature *sig)
 	result->this_arg.byref__ = TRUE;
 	result->blittable = TRUE;
 	result->inited = TRUE;
+	result->type_token = MONO_TOKEN_TYPE_DEF;
```

Comment:
Failures might be related to this change.

Reply:
Failures are related and two work items:
- JIT.opt `JIT/opt/ValueNumbering/TypeTestFolding/TypeTestFolding.dll`
- Loader.classloader `Loader/classloader/Casting/Functionpointer/Functionpointer.sh [FAIL]

are failing due to: 
```
src/mono/mono/metadata/metadata.c:1326, condition `GINT_TO_UINT32(idx) < table_info_get_rows (t)' not met
```
This is observed on: `linux-x64 Release AllSubsets_Mono_LLVMAot_RuntimeTests llvmaot` leg